README.md

```diff
@@ -1,21 +1,161 @@
+
+### Benefits of P2P
+
+- Direct communication: reduce latency and bandwidth.
+- No servers: no cloud, no scaling, no deploy, no devops...
+- No SaaS:
+  - Availability.
+  - No usage restriction.
+  - No unknown (malicious or not) usage of your data.
+
+### Added complexities of being fully distributed
+
+- Data consistency.
+- Network partition / topology.
+- Peer discovery and connection establishment.
+- Authentication / Authorization.
+- Security and privacy.
+
+We have a particular interest in P2P application development. You can read more in the [About](#-about) section.
+
+## 🏗️ Architecture
+
+![Architecture Diagram](assets/architecture_diagram.jpg)
+
+P2P Editor building blocks:
+- Code editor: [Ace](https://github.com/ajaxorg/ace)
+- Database: append only log ([hypercore](https://github.com/mafintosh/hypercore))
+- Communication: WebRTC RTCDataChannel ([webrtc-swarm](https://github.com/mafintosh/webrtc-swarm))
+- Peer discovery: WebRTC signaling server ([signalhub](https://github.com/mafintosh/signalhub))
+- Storage: RAM
+
+Workflow:
+
+1. Peers are discovered via signaling server.
+2. For each change made in the editor, a delta representing that change is appended to hypercore database.
+3. Hypercore database is constantly replicating between all peers via WebRTC RTCDataChannel.
+4. When a peer receives a delta it is applied to its editor instance.
+
+## 🚀 Environment setup
+
+### 🐳 Needed tools
+
+1. [Install Docker](https://www.docker.com/get-started) 
+2. Clone this project: `git clone https://github.com/CodelyTV/p2p-editor`
+3. Move to the project folder: `cd p2p-editor`
+
+### 🛠️ Environment configuration
+
+1. Copy the default environment variables: `cp .env.dist .env`
+2. Modify the environment variables if needed: `vim .env`
+
+### 🌍 Application execution
+
+1. Start the application with Docker Compose: `docker-compose up`
+2. Go to [http://localhost:8000](http://localhost:8000)
+
+### ✅ Tests execution
+
+1. Execute Unit tests: `docker-compose run p2p-editor test`
+
+## 💻 Live demo
+
+1. To start a new live code session open [http://p2p-editor.codely.tv](http://p2p-editor.codely.tv).
+2. Share the URL with the users you want to join the session.
+3. Start editing the file.
+ 
+That's all! All changes are instantly reflected in the connected peers editor.
+
+Happy P2P coding!
+
+## ℹ️ About
+
+We started this project in the context of a learning process with the aim of continuing to learn about P2P and put into practice what we learned by building a useful P2P application.
+
+During this process, all the learning resources that we found and consider useful are published in [awesome-p2p]("https://github.com/CodelyTV/awesome-p2p).
+
+The initial implementation of P2P Editor is based on [Mathias Buus](https://github.com/mafintosh) work. From here thank you very much for sharing so many free (libre) resources and knowledge.
```

Comment:
```suggestion
The initial implementation of P2P Editor is based on [Mathias Buus](https://github.com/mafintosh) work. In case you read this, thank you very much for sharing so many free (libre) resources and knowledge 😊
```